File: app/src/main/java/in/completecourse/questionbank/adapter/CompetitionUpdatesAdapter.java
package in.completecourse.questionbank.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import in.completecourse.questionbank.PDFActivity;
import in.completecourse.questionbank.R;
import in.completecourse.questionbank.model.UpdateItem;

public class CompetitionUpdatesAdapter extends RecyclerView.Adapter<CompetitionUpdatesAdapter.UpdatesViewHolder> {
    private final Context context;
    private final ArrayList<UpdateItem> updateItemsArrayList;

    public CompetitionUpdatesAdapter(Context context, ArrayList<UpdateItem> list){
        this.context = context;
        this.updateItemsArrayList = list;
    }

    @NonNull
    @Override
    public UpdatesViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View view =inflater.inflate(R.layout.competition_update_item, viewGroup, false);
        return new UpdatesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final UpdatesViewHolder updatesViewHolder, int i) {
        final UpdateItem updateItem = updateItemsArrayList.get(i);
        updatesViewHolder.titleText.setText(updateItem.getUpdateKaName());
        updatesViewHolder.descText.setText(updateItem.getUpdateKaDesc());
        updatesViewHolder.serialText.setText(updateItem.getSerialNumber());

        updatesViewHolder.titleText.setOnClickListener(v -> {
            if (updatesViewHolder.descText.getVisibility() == View.GONE){
                updatesViewHolder.descText.setVisibility(View.VISIBLE);
                updatesViewHolder.knowMoreText.setVisibility(View.VISIBLE);
            }else{
                updatesViewHolder.descText.setVisibility(View.GONE);
                updatesViewHolder.knowMoreText.setVisibility(View.GONE);
            }
        });

        updatesViewHolder.knowMoreText.setOnClickListener(v -> {
            Intent intent = new Intent(context, PDFActivity.class);
            intent.putExtra("url", updateItem.getUpdateKaLink());
            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return updateItemsArrayList.size();
    }

    class UpdatesViewHolder extends RecyclerView.ViewHolder {
        final TextView serialText, titleText, descText, knowMoreText;

        UpdatesViewHolder(@NonNull View itemView) {
            super(itemView);
            serialText = itemView.findViewById(R.id.text_serial_number);
            titleText = itemView.findViewById(R.id.textUpdateTitle);
            descText = itemView.findViewById(R.id.desc);
            knowMoreText = itemView.findViewById(R.id.know_more);
        }
    }
}

File: app/src/main/java/in/completecourse/questionbank/adapter/CompetitionUpdatesAdapter.kt
package `in`.completecourse.questionbank.adapter

import `in`.completecourse.questionbank.PDFActivity
import `in`.completecourse.questionbank.R
import `in`.completecourse.questionbank.model.UpdateItem
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class CompetitionUpdatesAdapter(private val context: Context, list: ArrayList<UpdateItem>) : RecyclerView.Adapter<CompetitionUpdatesAdapter.UpdatesViewHolder>() {
    private val updateItemsArrayList: ArrayList<UpdateItem> = list
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): UpdatesViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.competition_update_item, viewGroup, false)
        return UpdatesViewHolder(view)
    }

    override fun onBindViewHolder(updatesViewHolder: UpdatesViewHolder, i: Int) {
        val updateItem: UpdateItem = updateItemsArrayList[i]
        updatesViewHolder.titleText.text = updateItem.updateKaName
        updatesViewHolder.descText.text = updateItem.updateKaDesc
        updatesViewHolder.serialText.text = updateItem.serialNumber
        updatesViewHolder.titleText.setOnClickListener {
            if (updatesViewHolder.descText.visibility == View.GONE) {
                updatesViewHolder.descText.visibility = View.VISIBLE
                updatesViewHolder.knowMoreText.visibility = View.VISIBLE
            } else {
                updatesViewHolder.descText.visibility = View.GONE
                updatesViewHolder.knowMoreText.visibility = View.GONE
            }
        }
        updatesViewHolder.knowMoreText.setOnClickListener { v: View ->
            val intent = Intent(context, PDFActivity::class.java)
            intent.putExtra("url", updateItem.updateKaLink)
            v.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return updateItemsArrayList.size
    }

    class UpdatesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val serialText: TextView = itemView.findViewById(R.id.text_serial_number)
        val titleText: TextView = itemView.findViewById(R.id.textUpdateTitle)
        val descText: TextView = itemView.findViewById(R.id.desc)
        val knowMoreText: TextView = itemView.findViewById(R.id.know_more)

    }

}